<?xml version="1.0"?>
<robot name="Bobbis">

  <material name="blue">
    <color rgba="0 0 0.8 1"/>
  </material>
  <material name="black">
    <color rgba="0 0 0 1"/>
  </material>
  <material name="white">
    <color rgba="1 1 1 1"/>
  </material>
  <material name="purple">
    <color rgba="0.7 0 0.6 1"/>
  </material>
  <material name="red">
    <color rgba="1 0 0 1"/>
  </material>
  <material name="green">
    <color rgba="0 1 0 1"/>
  </material>
  <material name="yellow">
    <color rgba="0.7 0.2 0.2 1"/>
  </material>
  
  <link name="base_link">
    <visual>
      <geometry>
        <box size="0.4 0.4 0.1"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0.05"/>
      <material name="blue"/>
    </visual>
  </link>
 
   <link name="part_1">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.1" radius="0.05"/>
      </geometry>
      <material name="red"/>
    </visual>
  </link>
 
  <joint name="pivot_01" type="continuous">
    <parent link="base_link"/>
    <child link="part_1"/>
    <axis rpy="-1.57080 0 3.14159" xyz="0 0 1"/>
    <origin rpy="0.0 0.0 0.0" velocity="1.0 1.0 1.0"/>
  </joint>
  
  <link name="part_2">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0.13"/>
      <geometry>
        <cylinder length="0.3" radius="0.05"/>
      </geometry>
      <material name="purple"/>
    </visual>
  </link>
  
  <joint name="pivot_12" type="continuous">
    <parent link="part_1"/>
    <child link="part_2"/>
    <axis rpy="0 0 0" xyz="0 1 0"/>
    <origin rpy="0 0 0" xyz="0 0 0.05"/>
    <limit effort="0.0" velocity="1.0"/>
    
  </joint>
  
  <link name="part_3">
    <visual>
      <origin rpy="0 1.57080 0" xyz="0.125 0 0"/>
      <geometry>
        <cylinder length="0.25" radius="0.05"/>
      </geometry>
      <material name="white"/>
    </visual>
  </link>
  
  <joint name="pivot_23" type="continuous">
    <parent link="part_2"/>
    <child link="part_3"/>
    <axis rpy="0 0 0" xyz="0 1 0"/>
    <origin rpy="0 0 0" xyz="0 0 0.28"/>
    <limit effort="0.0" velocity="1.0"/>
  </joint>
  
  <link name="part_4">
    <visual>
      <origin rpy="0 1.57075 0" xyz="0.075 0 0"/>
      <geometry>
        <cylinder length="0.15" radius="0.03"/>
      </geometry>
      <material name="black"/>
    </visual>
  </link>
  
  <joint name="pivot_34" type="continuous">
    <parent link="part_3"/>
    <child link="part_4"/>
    <axis rpy="0 0 0" xyz="1 0 0"/>
    <origin rpy="0 0 0" xyz="0.25 0 0"/>
    <limit effort="0.0" velocity="1.0"/>
  </joint>
  
  <link name="part_5">
    <visual>
      <origin rpy="0 1.57075 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.03" radius="0.03"/>
      </geometry>
      <material name="green"/>
    </visual>
  </link>
  
  <joint name="pivot_45" type="continuous">
    <parent link="part_4"/>
    <child link="part_5"/>
    <axis rpy="0 0 0" xyz="0 1 0"/>
    <origin rpy="0 0 0" xyz="0.15 0 0"/>
    <limit effort="0.0" velocity="1.0"/>
  </joint>
  
  <link name="part_6">
    <visual>
      <origin rpy="0 1.57075 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.07" radius="0.03"/>
      </geometry>
      <material name="yellow"/>
    </visual>
  </link>
  
  <joint name="pivot_56" type="continuous">
    <parent link="part_5"/>
    <child link="part_6"/>
    <axis rpy="0 0 0" xyz="1 0 0"/>
    <origin rpy="0 0 0" xyz="0.03 0 0"/>
    <limit effort="0.0" velocity="1.0"/>
  </joint>
</robot>

